lib/iris/_concatenate.py

```diff
@@ -786,7 +787,7 @@ def _coordinate_differences(self, other, attr, reason="metadata"):
             diff_names = []
             for self_key, self_value in self_dict.items():
                 other_value = other_dict[self_key]
-                if self_value != other_value:
+                if hexdigest(self_value) != hexdigest(other_value):
```

Comment:
Note this routine is purely informational - it doesn't change any Iris behaviour.

This change should have been made years ago, since the behaviour it is providing information on is already using `hexdigest` (see `iris.common.metadata`).